content/en/blog/2019/alb-ingress-gateway-iks/index.md

```diff
@@ -0,0 +1,394 @@
+    metadata:
+      name: default
+    spec:
+      host: "*.httptools.svc.cluster.local"
+      trafficPolicy:
+        tls:
+          mode: ISTIO_MUTUAL
+    EOF
+    {{< /text >}}
+
+## Create secrets for ALB and Istio ingress gateway
+
+IKS generates a TLS certificate and a private key and stores them as a secret in the default namespace when you register
+a DNS domain for an external IP by the `ibmcloud ks nlb-dns-create` command. IKS stores the Application Load Balancer's
+certificate and private key also as a secret in the default namespace. You need these credentials to establish the
+identities the ALB and the Istio ingress gateway will present one to another during mutual TLS authentication between
+them. You configure the ALB and the Istio ingress gateway to exchange these certificates signed by their private keys,
+and to trust the certificates one of another.
+
+1.  Store the name of your cluster in the `CLUSTER_NAME` environment variable:
+
+    {{< text bash >}}
+    $ export CLUSTER_NAME=<your cluster name>
+    {{< /text >}}
+
+1.  Store the domain name of your ALB in the `ALB_INGRESS_DOMAIN` environment variable:
+
+    {{< text bash >}}
+    $ ibmcloud ks cluster get --cluster $CLUSTER_NAME | grep Ingress
+    Ingress Subdomain:              <your ALB ingress domain>
+    Ingress Secret:                 <your ALB secret>
+    {{< /text >}}
+
+    {{< text bash >}}
+    $ export ALB_INGRESS_DOMAIN=<your ALB ingress domain>
+    $ export ALB_SECRET=<your ALB secret>
+    {{< /text >}}
+
+1.  Store the external IP of your `istio-ingressgateway` service in an environment variable.
+
+    {{< text bash >}}
+    $ export INGRESS_GATEWAY_IP=$(kubectl -n istio-system get service istio-ingressgateway -o jsonpath='{.status.loadBalancer.ingress[0].ip}')
+    $ echo INGRESS_GATEWAY_IP = $INGRESS_GATEWAY_IP
+    {{< /text >}}
+
+1.  Create a DNS domain for the IP of the Istio Ingress Gateway service:
+
+    {{< text bash >}}
+    $ ibmcloud ks nlb-dns-create --cluster $CLUSTER_NAME --ip $INGRESS_GATEWAY_IP
+    Host name subdomain is created as <some domain>
+    {{< /text >}}
+
+1.  Store the domain name from the previous command in an environment variable:
+
+    {{< text bash >}}
+    $ export INGRESS_GATEWAY_DOMAIN=<the domain from the previous command>
+    {{< /text >}}
+
+1.  List the registered domain names:
+
+    {{< text bash >}}
+    $ ibmcloud ks nlb-dnss --cluster $CLUSTER_NAME
+    Retrieving host names, certificates, IPs, and health check monitors for network load balancer (NLB) pods in cluster <your cluster>...
+    OK
+    Hostname                          IP(s)                       Health Monitor   SSL Cert Status   SSL Cert Secret Name
+    <your ingress gateway hostname>   <your ingress gateway IP>   None             created           <the matching secret name>
+    ...
+    {{< /text >}}
+
+    Wait until the status of the certificate (the third field) in the line that matches the IP of the Istio ingress
```

Comment:
Believe this is actually the fourth field, not the third: 
```
        1.          2.               3.                        4.                            5.                                6. 
Hostname,IP(s),Health Monitor,SSL Cert Status,SSL Cert Secret Name,Secret Namespace 
```

Reply:
Fixed by https://github.com/istio/istio.io/pull/5098/commits/29a21c0cef9de5b6c20d1f4eeb555c1b40be7d81.